File: src/0496-next-greater-element-i/solution.go
func nextGreaterElement(n1 []int, n2 []int) []int {
    a := make([]int, len(n1), len(n1))
    for i := 0; i < len(a); i++ { a[i] = -1 }
    for i := 0; i < len(n1); i++ {
        k := 0
        for j := 0; j < len(n2); j++ {
            if n1[i] == n2[j] {
                k = j
                break
            }
        }
        for k := k; k < len(n2); k++ {
            if n1[i] < n2[k] {
                a[i] = n2[k]
                break
            }
        }
    }
    return a
}

File: src/0496-next-greater-element-i/solution.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        next_greater = {}
        stack = []
    
        for num in nums2:
            while stack and num > stack[-1]:
                next_greater[stack.pop()] = num
            stack.append(num)
    
        return [next_greater.get(num, -1) for num in nums1]
